src/data/mystery-encounters/encounters/weird-dream-encounter.ts

```diff
@@ -566,3 +544,83 @@ function doSideBySideTransformations(scene: BattleScene, transformations: Pokemo
     }
   });
 }
+
+/**
+ * Returns index of the new egg move within the Pokemon's moveset (not the index of the move in `speciesEggMoves`)
+ * @param scene
+ * @param newPokemon
+ * @param speciesRootForm
+ */
+async function addEggMoveToNewPokemonMoveset(scene: BattleScene, newPokemon: PlayerPokemon, speciesRootForm: Species): Promise<number | null> {
+  let eggMoveIndex: null | number = null;
+  if (speciesEggMoves.hasOwnProperty(speciesRootForm)) {
+    const eggMoves: Moves[] = speciesEggMoves[speciesRootForm].slice(0);
```

Comment:
NIT: 
There is [Pokemon#getEggMoves()](https://github.com/pagefaultgames/pokerogue/blob/abb8abcceff3a566c0f4506cd278f8574f88917e/src/field/pokemon.ts#L1797-L1804)

```suggestion
    const eggMoves = newPokemon.getEggMoves();
```